Sources/Datadog/RUM/RUMMonitor/Scopes/RUMApplicationScope.swift

```diff
@@ -123,4 +129,42 @@ internal class RUMApplicationScope: RUMScope, RUMContextProvider {
         let isDiscarded = !sessionScope.isSampled
         dependencies.onSessionStart?(sessionID, isDiscarded)
     }
+
+    private func sendTelemetryDebugEvent(command: RUMTelemetryDebugCommand) {
+        let dateCorrection = dependencies.dateCorrector.currentCorrection
+        let actionId = context.activeUserActionID?.toRUMDataFormat
+        let viewId = context.activeViewID?.toRUMDataFormat
+        let event = TelemetryDebugEvent(
+            dd: TelemetryDebugEvent.DD(),
+            action: actionId.flatMap { .init(id: $0) },
+            application: .init(id: context.rumApplicationID),
+            date: dateCorrection.applying(to: command.time).timeIntervalSince1970.toInt64Milliseconds,
+            service: "dd-sdk-ios",
+            session: .init(id: context.sessionID.toRUMDataFormat),
```

Comment:
The `context.sessionID` will be `00000000-0000-0000-0000-000000000000` if the session is not sampled. Also, this event will be sent no matter of session sampling as we write it to the output unconditionally. Is this what we expect and won't this zero-ed session ID be causing any problems on BE? 🤔

Reply:
Indeed, in such a case we should send a null session ID, I'll update the PR. Sending telemetry events when the session is sampled out is indeed something we want to keep (in order to detect issues even for unkept sessions)